src/main/java/com/oneops/api/resource/Transition.java

```diff
@@ -1889,6 +1988,96 @@ public CiResource updateRelay(String environmentName, String relayName, String s
+											 JSONObject jsonObject) throws OneOpsClientAPIException {
+		// Start a new deployment
+		Response response = doDeploy(environmentName, transitionEnvUri, request, jsonObject);
+		checkForErrors(environmentName, response, IConstants.EXISTING_DEPLOYMENT_MESSAGE);
+
+		// Wait for 5s before checking if the deployment started successfully
+		try {
+			Thread.sleep(5000);
+		} catch(InterruptedException ie) {
+			LOG.error("Error trying to delay: {}", ie.getMessage());
+		}
+
+		// Check for capacity issues and if deployment has started or failed due to capacity error.
+		response = doDeploy(environmentName, transitionEnvUri, request, jsonObject);
+		checkForErrors(environmentName, response, null);
+	}
+
+	private Response doDeploy(String environmentName, String transitionEnvUri,
+									 RequestSpecification request, JSONObject jsonObject) throws OneOpsClientAPIException {
+
+		Response response = request.body(jsonObject.toString()).post(transitionEnvUri + environmentName + "/deployments");
+		if(response == null) {
+			String msg = String.format("Failed to start deployment for environment %s " +
+					"due to null response" ,environmentName);
+			throw new OneOpsClientAPIException(msg);
+		}
+
+		return response;
+	}
+
+	/**
+	 * Errors during commit and deploy are part of comments of the Oneops API response,
+	 * unless its a server error (status code 500).
+	 * So we need to parse the comments for errors.
+	 * As we need to call the deploy API twice
+	 * 1st call - returns status if deployment plan is generated correctly
+	 *   a. If the comment contains an error that an active deployment is in progress,
+	 *   throw an exception
+	 * 2nd call - returns one of the following -
+	 *   a. deployment plan generated successfully
+	 *   b. An error that an active deployment already in progress
+	 *   c. Error starting deployment for any reason
+	 * (a) or (b) in comments indicate the deployment has started successfully
+	 * (c) indicates there was a problem with starting the deployment
+	 * So, we should ignore the message is (a) or (b), but throw an exception if its (c)
+	 *
+	 * @param environmentName
+	 * @param response
+	 * @param checkForErrorMessage
+	 * @throws OneOpsClientAPIException
+	 */
+
+	private void checkForErrors(String environmentName, Response response, String checkForErrorMessage) throws OneOpsClientAPIException {
+		if(response.getStatusCode() == 200 || response.getStatusCode() == 302) {
+			CiResource body = response.getBody().as(CiResource.class);
+			String comment = body.getComments();
+			if (comment.startsWith("ERROR:BOM:")) {
+				String[] commentParts = comment.split(":");
+
+				// When deployment API is called again to check if the deployment has started correctly
+				// ignore the active deployment error message.
+				// But if we are specifically checking for that error message, handle the error (1st Deploy API call)
+				if (commentParts[2].startsWith(IConstants.EXISTING_DEPLOYMENT_MESSAGE)
```

Comment:
I am not able to understand this logic. I believe, if we can avoid the second deployment api call (to check status) as mentioned in earlier comment, we could get rid of this and have a cleaner error check.

Reply:
Done